feat: Implement transformations logic.

marianogappa opened the pull request:
This is a first draft of sdk changes to add transformations support, so that transformer plugins can be created.

I don't want to merge it yet, because the channel logic is not well thought out. I'd rather want to do a test sync locally with some transformation.

marianogappa commented on internal/reversertransformer/reversertransformer_test.go:
```diff
@@ -0,0 +1,107 @@
+
+func TestReverserTransformer(t *testing.T) {
+	p := plugin.NewPlugin("test", "development", GetNewClient())
+	p.SetSkipTableValidation(true)
```
This is annoying 👀  a plugin doesn't know what it does, so it doesn't know that it has to automatically skip this validation if it's a transformer plugin. I don't like that whoever implements a Transformer plugin has to remember to skip this.

erezrokah reviewed (commented):
This looks good 🚀 Thank you for adding the tests 👍 

Will wait for an example plugin in our monorepo to approve

marianogappa commented:
I replaced this sdk onto the monorepo and added preliminary support for transformers in syncs, and I added a "reverser" transformer. Here's the result of hooking it up on an xkcd sync:

![Screenshot 2024-07-19 at 12 59 06](https://github.com/user-attachments/assets/910340c0-09b6-4b75-9b8f-5d3a867fd901)
